force_gromacs/data_sources/molecule/molecule_model.py

```diff
@@ -1,103 +1,58 @@
-from traits.api import Unicode, File
+from traits.api import Int, List, on_trait_change
 
-from force_bdss.api import BaseDataSourceModel
-from force_bdss.core.verifier import VerifierError
+from force_bdss.api import BaseDataSourceModel, VerifierError
 
 
 class MoleculeDataSourceModel(BaseDataSourceModel):
-    """Class containing all parameters for a single molecule
-    ingredient (molecular species) in a Gromacs
-    simulation"""
+    """Class containing all fragments to create for a single Gromacs
+    `Molecule` object"""
 
-    # --------------------
-    #  Required Attributes
-    # --------------------
+    #: Number of molecular fragments
+    n_fragments = Int(1,
+                      desc='Number of constituent fragments',
+                      changes_slots=True, verify=True)
 
-    #: Name of the molecule
-    name = Unicode(
-        desc='Name of molecule')
-
-    #: Reference symbol of molecule in Gromacs files
-    symbol = Unicode(
-        desc='Reference symbol in input Gromacs topology file')
-
-    #: Location of Gromacs topology file containing molecular data
-    topology = File(
-        desc='File path for Gromacs topology file',
-        verify=True
+    #: Stoichiometric numbers of each molecular fragment
+    fragment_numbers = List(
```

Comment:
`View` of this trait can be worked on - must be a fixed length (`n_fragments`), but we do not expect this to be greater than 10, so maybe a `TableEditor` could come in handy?